odoo/models.py

```diff
@@ -3072,7 +3072,10 @@ def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query:
 
                 if params:
                     if fname != 'id':
-                        params = [field.convert_to_column(p, self, validate=False) for p in params]
+                        if field.translate:
+                            params = [field.convert_to_column(p, self, validate=False).adapted['en_US'] for p in params]
+                        else:
+                            params = [field.convert_to_column(p, self, validate=False) for p in params]
```

Comment:
Given that those expressions are very much alike, I suggest to factor out their common part as:
```suggestion
                        params = [field.convert_to_column(p, self, validate=False) for p in params]
                        if field.translate:
                            # unwrap actual values from the Json values returned by convert_to_column()
                            params = [p.adapted['en_US'] for p in params]
```